reference-architecture/azure-ansible/bastion.sh

```diff
@@ -0,0 +1,345 @@
+  - name: add new initiator name
+    lineinfile: dest=/etc/iscsi/initiatorname.iscsi create=yes regexp="InitiatorName=*" line="InitiatorName=iqn.2016-02.local.azure.nodes" state=present
+  - name: restart iscsid service
+    shell: systemctl restart iscsi
+    ignore_errors: yes
+  - name: Enable Iscsi
+    shell: systemctl enable iscsi
+    ignore_errors: yes
+  - name: Start iScsi Initiator  Service
+    shell: systemctl start iscsi
+    ignore_errors: yes
+  - name: Discover Devices on Iscsi  All Hosts
+    shell: iscsiadm --mode discovery --type sendtargets --portal store1
+    register: task_result
+    until: task_result.rc == 0
+    retries: 10
+    delay: 30
+    ignore_errors: yes
+  - name: Login All Hosts
+    shell: iscsiadm --mode node --portal store1 --login
+    register: task_result
+    until: task_result.rc == 0
+    retries: 10
+    delay: 30
+    ignore_errors: yes
+EOF
+
+
+cat <<EOF > /home/${AUSERNAME}/postinstall.yml
+---
+- hosts: masters
+  vars:
+    description: "auth users"
+  tasks:
+  - name: Create Master Directory
+    file: path=/etc/origin/master state=directory
+  - name: add initial user to OpenShift Enterprise
+    shell: htpasswd -c -b /etc/origin/master/htpasswd ${AUSERNAME} ${PASSWORD}
+
+EOF
+
+cat <<EOF > /home/${AUSERNAME}/openshift-install.sh
+export ANSIBLE_HOST_KEY_CHECKING=False
+sleep 120
+ansible all --module-name=ping > ansible-preinstall-ping.out || true
+ansible-playbook  /home/${AUSERNAME}/subscribe.yml
+echo "${RESOURCEGROUP} Bastion Host is starting ansible BYO" | mail -s "${RESOURCEGROUP} Bastion BYO Install" ${RHNUSERNAME} || true
+ansible-playbook  /usr/share/ansible/openshift-ansible/playbooks/byo/config.yml < /dev/null &> byo.out
+
+# ssh gwest@master1 oadm registry --selector=region=infra
+# ssh gwest@master1 oadm router --selector=region=infra
+wget http://master1:8443/api > healtcheck.out
+ansible-playbook /home/${AUSERNAME}/quota.yml
+ansible-playbook /home/${AUSERNAME}/postinstall.yml
+ansible-playbook /home/${AUSERNAME}/setupiscsi.yml
+cd /root
+mkdir .kube
+scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ${AUSERNAME}@master1:~/.kube/config /tmp/kube-config
+cp /tmp/kube-config /root/.kube/config
+mkdir /home/${AUSERNAME}/.kube
+cp /tmp/kube-config /home/${AUSERNAME}/.kube/config
+chown --recursive ${AUSERNAME} /home/${AUSERNAME}/.kube
+rm -f /tmp/kube-config
+ansible-playbook /home/${AUSERNAME}/setupiscsi.yml
+echo "${RESOURCEGROUP} Installation Is Complete" | mail -s "${RESOURCEGROUP} Install Complete" ${RHNUSERNAME} || true
+EOF
+
+cat <<EOF > /home/${AUSERNAME}/.ansible.cfg
+[defaults]
+remote_tmp     = ~/.ansible/tmp
+local_tmp      = ~/.ansible/tmp
+host_key_checking = False
+forks=30
+gather_timeout=60
+timeout=240
+[ssh_connection]
+control_path = ~/.ansible/cp/ssh%%h-%%p-%%r
+ssh_args = -o ControlMaster=auto -o ControlPersist=600s -o ControlPath=~/.ansible/cp-%h-%p-%r
+EOF
+chown ${AUSERNAME} /home/${AUSERNAME}/.ansible.cfg
```

Comment:
True, but considering the way these are "parsed" has changed quite a bit, I'd say it needs to be in several places for now. As I ran into issues with ansible not reading parameters. Actually this whole issue, is there default, buried deep in ansible was not set properly. There already a "bug" on it upstream, but I didnt see the timing when it would "flow" downwards, so till next "ansible", this was a work around. (And over 2 weeks of time tracking what the cause is.